addons/html_builder/static/src/builder/snippet_model.js

```diff
@@ -78,6 +78,7 @@ export class SnippetModel extends Reactive {
                     isCustom: false,
                     imagePreviewSrc: snippetEl.dataset.oImagePreview,
                 };
+                snippetEl.children[0].dataset["name"] = snippet.title;
```

Comment:
The snippet that is inserted on the DOM is `snippetEl.children[0]`. As I needed the `title` information to display the correct name on the invisible panel, I put this information on the DOM so that it is accessible to the  `InvisibleElementsPanel` componenent. What we could have done, is to put the `SnippetModel` in the `env` so that all the components have access to its information. However, we would still need to put an information on the DOM in order to correctly link a snippet on the DOM to the `SnippetModel` data. As it seems that the only useful information is the title, I put it on the DOM to stay close to the current implementation in website. However, I agree with you that this is not the best place to do it; it should probably be done by the py. Maybe I can add a TODO comment for the moment to change it later as this is the way its done in website now?